lib/trees/abstract_tree.nit

```diff
@@ -36,13 +36,19 @@ abstract class TreeMap[K: Comparable, E]
 	fun show_dot is abstract
 end
 
-# Node used in Tree implementation
-# nodes are used to store values
-# * `E`: type of value
-class TreeNode[K: Comparable, E]
+# Abstract node structure used in `TreeMap` implementations
+#
+# Nodes are defined recursively each node (except the root one) pointing to a parent.
+# Nodes can be used to store data with the `AbstractTreeNode::value` attribute.
+#
+# Formal parameters:
+# * `K`: key type (a `Comparable` one so nodes can be sorted by their keys)
+# * `E`: value type (to store your data)
+abstract class AbstractTreeNode[K: Comparable, E]
```

Comment:
In this case I will need a better name idea :)

I renamed this one so I can use `TreeNode` for the kind that goes in a `Tree`. Maybe `TreeMapNode` for the abstract one?